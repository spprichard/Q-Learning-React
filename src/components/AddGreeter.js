import React, {Component} from 'react'

class AddGreeter extends Component {
    constructor(props){
        super(props);
        this.handleUpdate = this.handleUpdate.bind(this);
        this.addGreeting = this.addGreeting.bind(this);
        this.state = {
            greetingName: ''
        }
    }

    addGreeting(){
        this.props.addGreeting(this.state.greetingName);
        this.setState({
            greetingName: ''
        });
    }

    handleUpdate(event){
        this.setState({
            greetingName : event.target.value

        });
    }

    render(){
        return (
            <div className = "AddGreeter"> 
                <p className="Title">Enter your name:</p>
                <input type="text" onChange={this.handleUpdate} value={this.state.greetingName}/>
                &nbsp;&nbsp;
                <button onClick={this.addGreeting}>Add</button>
            </div>
        );
    }
}

export default AddGreeter;