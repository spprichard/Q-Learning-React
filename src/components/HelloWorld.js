import React, { Component } from 'react';

class HelloWorld extends Component {
  constructor(props){
    super(props);
    this.frenchify = this.frenchify.bind(this)
    this.resetGreeting = this.resetGreeting.bind(this)
    this.state = {
      greeting: "Hello "
    };
    
  }

  frenchify(){
    this.setState({
      greeting: "Bonjur "
    });
  }

  resetGreeting(){
    this.setState({
      greeting: "Hello "
    });
  }

  render() {
    return (
      <div className="HelloWorld">
        <h1>{this.state.greeting} {this.props.name}!</h1>
        <br/>
        <button className="RegularButton" onClick={this.frenchify}>Frenchify!</button>
        <button className="ResetButton" onClick={this.resetGreeting}>Reset</button>
      </div>
    );
  }
}

export default HelloWorld;
