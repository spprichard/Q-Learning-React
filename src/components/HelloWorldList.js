import React, { Component } from 'react';
import HelloWorld from "./HelloWorld"

class HelloWorldList extends Component{
    render(){
        return(
        <div className="HelloWorldList">
            <HelloWorld name="Tess"/>
            <HelloWorld name="Steven"/>
        </div>
        );
    }
}

export default HelloWorldList;