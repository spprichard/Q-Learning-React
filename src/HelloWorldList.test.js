import React from 'react';
import { shallow, render } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorldList from './components/HelloWorldList';
import HelloWorld from './components/HelloWorld';
import AddGreeter from './components/AddGreeter';

describe(HelloWorld, () => {
    const component = shallow(
        <HelloWorldList />
    );

    // snapshot test
    it("renders and matches out snapshot", () => {
        const component = renderer.create(
            <HelloWorldList />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("contains an AddGreeter subcompenent", () => {
        expect(component.find(AddGreeter)).toHaveLength(1);
    });

    it("contains the same number of HelloWorld components as greetings", () => {
        const helloWorlds = component.find(HelloWorld).length;
        const greetings = component.state('greetings').length;
        expect(helloWorlds).toEqual(greetings);
    });

    it("adds another greeting when add greeting function is called", () => {
        const before = component.find(HelloWorld).length;
        component.instance().addGreeting("Sample");
        component.update();
        const after = component.find(HelloWorld).length
        expect(after).toBeGreaterThan(before);
    });

    it("removes greeting when remove greeting function is called", () => {
        const before = component.find(HelloWorld).length;
        component.instance().removeGreeting("Sample")
        component.update();
        const after = component.find(HelloWorld).length
        expect(after).toBeLessThan(before);
    });
});