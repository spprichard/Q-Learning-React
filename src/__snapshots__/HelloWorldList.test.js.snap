// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorld extends _react.Component {
  constructor(props) {
    super(props);
    this.frenchify = this.frenchify.bind(this);
    this.resetGreeting = this.resetGreeting.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
    this.state = {
      greeting: "Hello " };


  }

  frenchify() {
    this.setState({
      greeting: "Bonjour " });

  }

  resetGreeting() {
    this.setState({
      greeting: "Hello " });

  }

  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }

  render() {
    return (
      _react2.default.createElement("div", { className: "HelloWorld", __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this },
        _react2.default.createElement("h1", { __source: { fileName: _jsxFileName, lineNumber: 34 }, __self: this }, this.state.greeting, " ", this.props.name, "!"),
        _react2.default.createElement("br", { __source: { fileName: _jsxFileName, lineNumber: 35 }, __self: this }),
        _react2.default.createElement("button", { className: "FrenchifyButton", onClick: this.frenchify, __source: { fileName: _jsxFileName, lineNumber: 36 }, __self: this }, "Frenchify!"),
        _react2.default.createElement("button", { className: "ResetButton", onClick: this.resetGreeting, __source: { fileName: _jsxFileName, lineNumber: 37 }, __self: this }, "Reset"),
        _react2.default.createElement("button", { className: "RemoveButton", onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 38 }, __self: this }, "Remove Me")));


  }} renders and matches out snapshot 1`] = `
<div
  className="HelloWorldList"
>
  <div
    className="AddGreeter"
  >
    <p
      className="Title"
    >
      Enter your name:
    </p>
    <input
      className="Input"
      onChange={[Function]}
      type="text"
      value=""
    />
      
    <button
      className="AddButton"
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
